File: src/dsa/BuySell.java
package dsa;

public class BuySell {
    public static int getMaxProfit(int[] stockPrices) {
        if (stockPrices.length < 2)
            throw new IllegalArgumentException("Getting a profit requires at least 2 prices.");

        int minPrice = stockPrices[0];
        int maxProfit = stockPrices[1] - stockPrices[0];

        for (int i = 1; i < stockPrices.length; i++) {
            int currentPrice = stockPrices[i];
            int potentialProfit = currentPrice - minPrice;
            maxProfit = Math.max(maxProfit, potentialProfit);
            minPrice = Math.min(minPrice, currentPrice);
        }
        return maxProfit;
    }

    public static void main(String[] args) {
        int[] stockPrices = {10, 7, 5, 8, 11, 9};
        int maxProfit = getMaxProfit(stockPrices);
        System.out.println(maxProfit);
    }
}

File: src/dsa-py/BuySell.py
def get_max_profit(stock_prices):
    if len(stock_prices) < 2:
        return "Getting a profit requires at least 2 prices."

    min_price = stock_prices[0]
    max_profit = stock_prices[1] - stock_prices[0]

    for i in range(len(stock_prices)):
        current_price = stock_prices[i]
        potential_profit = current_price - min_price
        max_profit = max(max_profit, potential_profit)
        min_price = min(min_price, current_price)

    return max_profit


stock_prices = [10, 7, 5, 8, 11, 9]
max_profit = get_max_profit(stock_prices)
print(max_profit)
